### PR TITLE
Fix issue DPTP-3132 Add HyperShift SA

### PR DESCRIPTION
**Pull request by bear-redhat**
Add the SA for connecting to our HyperShift deployment on Hive.

Ref: https://github.com/openshift/release/pull/33635

### CODE REVIEW

File: clusters/hive/hypershift.yaml [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    namespace: clusters
+    name: hypershift-workload-sa
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: hypershift-workload-rolebinding-admin
+roleRef:
+  kind: ClusterRole
+  name: cluster-admin
```
[the remark this answers is not in the document]
Comment (bear-redhat) [reply clusters/hive/hypershift.yaml:92]: Yes, at lease for now, until we figure out a better solution. The reason is that it need access to `cluster-XXX` namespace, which is dynamically generated for each cluster.

[the remark this answers is not in the document]
Comment (hongkailiu) [reply clusters/hive/hypershift.yaml:92]: /hold

The SA seems exposed to the test.
It seems dangerous to me.

File: clusters/hive/hypershift.yaml [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+  namespace: kube-public
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: hypershift-oidc-discovery
+subjects:
+  - kind: ServiceAccount
+    namespace: clusters
+    name: hypershift-workload-sa
+---
```
Comment (hongkailiu) [inline clusters/hive/hypershift.yaml:85]: Could you remove those? (Starting from line 85)

**Review by hongkailiu — APPROVED** [review]

**Pull request merged**